test/near_provider.test.js

```diff
@@ -359,6 +360,60 @@ describe('\n---- PROVIDER ----', () => {
             }));
         });
 
+        describe('retrieveNear | eth_retrieveNear', () => {
+            let account, value, firstNearBalance
+
+            beforeEach(withWeb3(async (web) => {
+                firstNearBalance = await web._provider.account.getAccountBalance()
+                account = utils.nearAccountToEvmAddress(ACCOUNT_ID)
+                value = 6 * 10 ** 18
+                await web.eth.sendTransaction({
+                    from: account,
+                    to: account,
+                    value: value.toString(),
+                    gas: 0
+                });
+            }));
+
+            test.only('sends near back to nearAccount if sufficient funds in corresponding evm account', withWeb3(async (web) => {
+                let valueRetrieved = value / 2;
+                let prevEvmBalance = new bn(await web.eth.getBalance(account, 'latest'))
+                let prevNearBalance = await web._provider.account.getAccountBalance()
+
+                console.log("addNear: diff total: ", new bn(firstNearBalance.total).sub(new bn(prevNearBalance.total)).toString())
+                console.log("addNear: diff available: ", new bn(firstNearBalance.available).sub(new bn(prevNearBalance.available)).toString())
+
+                let retrieveNear = await web.eth.retrieveNear({
+                    from: account,
+                    value: valueRetrieved,
+                    to: ACCOUNT_ID,
+                    gas: 0
+                })
+
+                let newEvmBalance = new bn(await web.eth.getBalance(account, 'latest'))
+                let newNearBalance = await web._provider.account.getAccountBalance()
+
+                console.log("retrieveNear: diff total: ", new bn(prevNearBalance.total).sub(new bn(newNearBalance.total)).toString())
+                console.log("retrieveNear: diff available: ", new bn(prevNearBalance.available).sub(new bn(newNearBalance.available)).toString()) 
```

Comment:
added some logging in trying to sort our near balances. I'm getting (somewhat imprecise?) changes in the near balance on `add_near` but no difference on `retrieve_near`. We tested this in the rust directory though and things checked out there. log output:

```
    console.log
      addNear: diff total:  6035994857667910000

      at test/near_provider.test.js:383:25

    console.log
      addNear: diff available:  6035994857667910000

      at test/near_provider.test.js:384:25

    console.log
      [evm]: evm log retrieve_near: Balance(3000000000000000000)

      at Account.printLogs (node_modules/near-api-js/lib/account.js:58:21)

    console.log
      retrieveNear: diff total:  0

      at test/near_provider.test.js:396:25

    console.log
      retrieveNear: diff available:  0

      at test/near_provider.test.js:397:25
```

Reply:
Is this something that needs to be resolved before merging?

Reply:
I kind of want to make an issue for it instead. I felt pretty blocked on this, might need outside help, and this PR is the bottleneck for the last 2 functions we need to add before everything is just cleaning up.

Reply:
K. Marking this conversation as resolved for this PR, and I'll let you make an issue for this.